Git Files/Git Config.sublime-syntax

```diff
@@ -0,0 +1,230 @@
+    # are interpreted as `t` by the git config parser, not TAB or "\t"
+    # (two chars). Therefore, we discourage the use of `\`.
+    - match: \\
+      scope: invalid.unnecessary-escape.git.config
+    - include: illegal-line-end-pop
+
+  # [section.subsection]
+  subsection-name-unquoted:
+    - match: '[\w-]+'
+      scope: string.unquoted.git.config
+    - match: \.
+      scope: punctuation.accessor.dot.git.config
+    - match: (?=\])
+      pop: true
+    - include: illegal-line-end-pop
+    - match: \S
+      scope: invalid.illegal.subsection-name.git.config
+
+  expect-section:
+    - match: (?=^\s*\[) # start of new section
+      pop: true
+    - match: \]
+      scope: punctuation.definition.brackets.end.git.config invalid.illegal.stray-bracket.git.config
+
+##[ SECTION BODY ]#####################################################
+
+  # changed = red
+  # untracked = bold green
+  key-color-pair:
+    - match: ^(\s*)({{variable_name}})(\s*(\=)\s*)
+      captures:
+        1: meta.mapping.git.config
+        2: meta.mapping.key.git.config variable.other.readwrite.git.config
+        3: meta.mapping.git.config
+        4: keyword.operator.assignment.git.config
+      push:
+        - meta_content_scope: meta.mapping.value.git.config
+        - include: color-value
+        - include: line-end
+    - include: expect-section
+
+  # key = val
+  key-value-pair:
+    - match: ^(\s*)({{variable_name}})(\s*(\=)\s*)
+      captures:
+        1: meta.mapping.git.config
+        2: meta.mapping.key.git.config variable.other.readwrite.git.config
+        3: meta.mapping.git.config
+        4: keyword.operator.assignment.git.config
+      push:
+        - meta_content_scope: meta.mapping.value.git.config
+        - include: other-value
+        - include: line-end
+    - include: expect-section
+
+##[ VALUES ]###########################################################
+
+  color-value:
+    # example: bold, italic, underline
+    - match: \b(?:no-?)?(?:ul|strike|reverse|italic|dim|bold|blink)\b
+      scope: support.constant.color-attribute.git.config
+    # example: red, blue, green
+    - match: \b(?:yellow|white|red|normal|magenta|green|cyan|blue|black|auto)\b
+      scope: support.constant.color.git.config
+    # example: 0-255
+    - match: \b(?:{{zero_to_255}})\b
+      scope: constant.other.color.rgb-value.git.config
+
+  other-value:
+    - include: boolean
+    - include: shell-script
+    - include: string-quoted
+    - include: string-unquoted
+
+  boolean:
+    # 0 and 1 are omitted due to mismatches (e.g. "log -1", "HEAD~1"). They will
+    # be matched as unquoted strings instead.
+    - match: \b(?:yes|true|on|off|no|false)\b
+      scope: constant.language.git.config
+
+  shell-script:
+    - match: (\")(\!)
+      captures:
+        1: string.quoted.double.git.config punctuation.definition.string.begin.git.config
+        2: keyword.control.import.shell.git.config
+      escape: \"(?=[^"]*(?:[#;]|$))
```

Comment:
It's the escape pattern to blame for the issue. It somehow needs to capture the last available `"` within the value without knowing about the internal structure/stack of the underlying shellscript syntax. It is always captured as prototype within the whole stack of shellscript breaking its stack logic. 

Modifying the line as follows: `escape: \"(?=[^"]+(?:[#;].*)?$)` will fix the provided examples. But if someone writes a real multiline script with double-quoted strings in the middle of the blocks the first occurance of `"` is handled as end-of-script.

Example:

```
  double_quote = "!echo \"The last quotation is broken.\";
    echo \"This line is broken.\";" # with quoted "comment"
```